semantic-analysis/src/Analysis/Effect/Domain.hs

```diff
@@ -71,7 +69,7 @@ dfalse = dbool False
 dbool :: Has (Dom val) sig m => Bool -> m val
 dbool = send . DBool
 
-dif :: Has (Dom val) sig m => val -> m a -> m a -> m a
+dif :: Has (Dom val) sig m => val -> m val -> m val -> m val
```

Comment:
This is necessary for us to collect the results from both branches of a conditional in exception tracking. `a` would always be `val` during abstract interpretation anyway, so it doesn't hurt us any.